crates/isograph_schema/src/validate_schema.rs

```diff
@@ -680,9 +716,14 @@ fn validate_client_field_definition_selections_exist_and_types_match(
                 schema_data,
                 server_fields,
                 client_field_args,
+                &mut used_variables,
             )
         })
-        .collect::<Result<_, _>>()
+        .collect::<Result<_, _>>()?;
+
+    validate_all_variables_are_used(variable_definitions, used_variables)?;
```

Comment:
#135 added a task to return multiple errors from this function.
It doesn't really matter for watch mode, but it is nice to see all the errors in VSCode when we work on the language server

Reply:
This is something I would like to do later on, maybe after I ran out of issues, we talked about in the last video call